### PR TITLE
Fixes to solve pthread build errors.

### PR DESCRIPTION
**Pull request by decada-robotics**
Some edits to prevent build errors in the rmf_ros2 package.

### CODE REVIEW

**Comment by mxgrey** [comment]
Please check if this other fix in `rmf_traffic` works for you: https://github.com/open-rmf/rmf_traffic/pull/74

**Comment by decada-robotics** [comment]
> What platform (OS + compiler) are you building on?

@codebot Using Ubuntu 20.04, colcon build. Building was fine on other similar devices but was not working for our desktop in particular.

File: rmf_fleet_adapter/CMakeLists.txt [hunk 1]
```diff
@@ -262,6 +262,7 @@ target_link_libraries(experimental_lift_watchdog
     ${rclcpp_LIBRARIES}
     ${rmf_fleet_msgs_LIBRARIES}
     ${std_msgs_LIBRARIES}
+    Threads::Threads
```
Comment (Yadunund) [inline rmf_fleet_adapter/CMakeLists.txt:265]: Looking at https://github.com/open-rmf/rmf_traffic/pull/74/files, can you also include a `find_package(Threads)`?

File: rmf_task_ros2/include/rmf_task_ros2/bidding/Response.hpp [hunk 2]
```diff
@@ -22,6 +22,7 @@
 #include <rmf_task_msgs/msg/bid_notice.hpp>
 #include <rmf_task_msgs/msg/bid_proposal.hpp>
 #include <rmf_task_msgs/msg/bid_response.hpp>
+#include <optional>
```
Comment (Yadunund) [inline rmf_task_ros2/include/rmf_task_ros2/bidding/Response.hpp:25]: Let's leave this unrelated change for a separate PR.

**Review by Yadunund — CHANGES_REQUESTED** [review]

File: rmf_task_ros2/include/rmf_task_ros2/bidding/Response.hpp (hunk 2 above)
Comment (samuelrawrs) [reply rmf_task_ros2/include/rmf_task_ros2/bidding/Response.hpp:25]: Hi, just wanted to add some details for this addition:

Some of our computers were having problems building without this include line.
This was the error that would show up: `'optional' in namspace 'std' does not name a template type`

after the addition of the include it built fine!

Comment (Yadunund) [reply rmf_task_ros2/include/rmf_task_ros2/bidding/Response.hpp:25]: I see, thanks for clarifying.

**Review by Yadunund — DISMISSED** [review]

**Review by mxgrey — APPROVED** [review]
